tsl/src/continuous_aggs/create.c

```diff
@@ -722,6 +726,37 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 						 errmsg(
 							 "time bucket function must reference a hypertable dimension column")));
 
+			if (list_length(fe->args) == 3)
+			{
+				/*
+				 * Currently the third argument of the bucketing function can be
+				 * only a timezone. Only immutable expressions can be specified.
+				 */
+				tz_arg = eval_const_expressions(NULL, lthird(fe->args));
+				if (IsA(tz_arg, Const))
+				{
+					Const *tz = castNode(Const, tz_arg);
+					Assert(tz->consttype == TEXTOID);
+					const char *tz_name = TextDatumGetCString(tz->constvalue);
```

Comment:
What if this is not `TEXTOID` in a production build? Won't it crash by calling `TextDatumGetCString()` on a different data type?

Reply:
This can't happen, is_valid_bucketing_function() will take care of it. I added the corresponding comment to clarify this.